internal/configuration/curves.go

```diff
@@ -17,8 +17,9 @@ type LinearCurveConfig struct {
 	Min int `json:"min"`
 	// Max is the maximum temperature in degrees
 	Max int `json:"max"`
-	// Steps is a map of temperature to PWM value
-	Steps map[int]float64 `json:"steps"`
+	// Steps is a map of temperature to relative speed value (in range of 0..255 or alternatively 0%..100%)
+	Steps      map[int]string
+	FloatSteps map[int]float64 `json:"steps"` // TODO: what is `json:"steps"` used for? should it be here or on Steps?
```

Comment:
`json:"steps"` specifies the name of the property in JSON parsing, which is used for both the API as well as the YAML config. Although if this is already working it seems like it is not used for the Config? :thinking:

Can you change the `fan2go.yaml` to use percent values to check?
I guess this should be the "preferred" way from now on anyway.
We should definitely add tests for this though anyway.

Reply:
That's what I assumed - but in fact the name in the yaml.config seems to depend on the literal variable name - `steps:` sets the `Steps` variable.

I was very confused by this, but the only way to get the values from the yaml to show up here was to name the new string map `Steps` (no matter where I put that `json:"steps"` thing) - originally I wanted to leave the float values in `Steps` and call the string one `StringSteps` or something, but that didn't work

Reply:
Apparently [viper](https://github.com/spf13/viper) uses [mapstructure](https://github.com/go-viper/mapstructure?tab=readme-ov-file), which has its own tags:

https://pkg.go.dev/github.com/go-viper/mapstructure/v2#hdr-Renaming_Fields

Reply:
Ok, so 
```
InSteps map[int]string `mapstructure:"user"`
Steps    map[int]float64 `json:"steps"`
```
should work?
(I guess the API should access the steps as floats?)

Reply:
Maybe like this?
```golang
Steps      map[int]string      `mapstructure:"steps" json:"-"`
FloatSteps map[float64]float64 `mapstructure:"-" json:"steps"`
```

The "minus" syntax is mentioned here:
![image](https://github.com/user-attachments/assets/212d12ca-5e53-475e-84f2-de009f2bb167)

Not sure if this also works for the `json` tag, but something similar probably exists.

Reply:
Thanks!

json also seems to use "-": https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go#ignoring-private-fields